src/main/java/http/ResponseHandler.java

```diff
@@ -1,9 +1,16 @@
 package http;
 
 public class ResponseHandler {
+    private static final Router router = new Router();
```

Comment:
SOLID Consideration: Initialising Router here couples it to ResponseHandler, see if you can lift the dependency.

Reply:
Lifts the dependency to the `ClientHandler` and injects the `Router` to the `ResponseHandler` in the latest commit.